packages/react-textarea/Spec.md

```diff
@@ -2,62 +2,160 @@
+  - `Outline`: TextArea has a white background, outline, and a line on the bottom.
+- Behavior variants
+  - `Disabled`: TextArea has a gray background, gray text, outline, and interaction is disabled.
+  - `Read Only`: TextArea has a gray background, gray outline, and typing is disabled.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### Component props:
+
+```ts
+type TextAreaProps = {
+  /**
+   * Which direction the TextArea is allowed to be resized.
+   *
+   * @defaultvalue none
+   */
+  resize?: 'none' | 'horizontal' | 'vertical' | 'both';
+
+  /**
+   * Size of the TextArea.
+   *
+   * @defaultvalue medium
+   */
+  size?: 'small' | 'medium' | 'large';
+
+  /**
+   * Styling the TextArea should use.
+   *
+   * @defaultvalue outline
+   */
+  appearance?: 'outline' | 'underline' | 'filledDarker' | 'filledLighter';
+
+  /**
+   * The value of the TextArea.
+   */
+  value?: string;
+
+  /**
+   * The default value of the TextArea.
+   */
+  defaultValue?: string;
+
+  /**
+   * Callback for when the user changes the value.
+   */
+  onChange?: (ev: React.FormEvent<HTMLTextAreaElement>, data: TextAreaOnChangeData) => void;
+};
+
+type TextAreaOnChangeData = {
+  value: string;
+};
+```
+
+The auto-resize feature is known to be unstable from v8 and tricky to implement. [As mentioned by ecraig12345](https://github.com/microsoft/fluentui/pull/21898#discussion_r816418917), we've been trying prioritize getting the components ready for production and may have to leave out some features for the initial version.
+
+`TextArea` will have a single root slot that will be `<textarea/>`.
+
+The design spec mentions a character count, error message, and label which will be left out for now.
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+**Public**
 
-## Migration
+```tsx
+<TextArea
+  className="textAreaClassName"
+  style={{ background: 'blue' }}
+  id="textarea-1"
+  value="This is a TextArea"
+  onChange={(ev, data) => console.log(data.value)}
+/>
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+**Internal**
 
-- _Migration from v8_
-- _Migration from v0_
+```tsx
+<slots.root {...slotProps.root}></slots.root>
+```
+
+**DOM** - how the component will be rendered as HTML elements
+
+```html
+<textarea id="textarea-1" style="background: blue" className="textAreaClassName" value="This is a TextArea"></textarea>
+```
+
+## Migration
+
+See [MIGRATION.md](MIGRATION.md).
 
 ## Behaviors
 
 _Explain how the component will behave in use, including:_
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+- Component States
+  - Rest
+  - Hover
+  - Pressed: Apply focus border animation.
+  - Focussed: Apply thick blue line on the bottom border.
+
+Interaction will be handled by native element.
```

Comment:
Since this part seems to be missing we should probably add a TODO here or open a separate issue to go back and add it.